File: patterns-java/src/test/java/me/freedom4live/patterns/java/prototype/IdealPersonTest.java
package me.freedom4live.patterns.java.prototype;

import org.junit.jupiter.api.Test;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.*;

class IdealPersonTest {

    @Test
    void clone_random_theSameObject() throws CloneNotSupportedException {
        for (int i = 0; i < 1000; i++) {
            //arrange
            IdealPerson person = new IdealPerson(UUID.randomUUID().toString());

            //act
            IdealPerson clonedPerson = person.clone();

            //assert
            assertNotNull(clonedPerson);
            assertEquals(person.getName(), clonedPerson.getName());
            assertNotSame(person, clonedPerson);
        }
    }

}

File: patterns-kotlin/src/test/kotlin/me/freedom4live/patterns/kotlin/prototype/IdealPersonTest.kt
package me.freedom4live.patterns.kotlin.prototype

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.util.*

internal class IdealPersonTest {

    @Test
    fun randomObject_clone_notTheSame() {
        repeat(1000) {
            //arrange
            val person = IdealPerson(UUID.randomUUID().toString())

            //act
            val clone = person.clone()

            //assert
            assertNotNull(clone)
            assertEquals(person.name, clone.name)
            assertNotSame(person, clone)
        }
    }

}
